scvid/train/training_plan.py

```diff
@@ -84,14 +84,15 @@ def configure_optimizers(self) -> dict[str, Any]:
             optim_config["lr_scheduler"] = {"scheduler": scheduler, "interval": "step"}
         return optim_config
 
-    def on_train_epoch_start(self):
+    def on_train_epoch_start(self) -> None:
         """
         Calls the ``set_epoch`` method on the iterable dataset of the given dataloader.
 
         If the dataset is ``IterableDataset`` and has ``set_epoch`` method defined, then
         ``set_epoch`` must be called at the beginning of every epoch to ensure shuffling
         applies a new ordering. This has no effect if shuffling is off.
         """
+        assert self.trainer.fit_loop._combined_loader is not None
         dataloaders = self.trainer.fit_loop._combined_loader.flattened
```

Comment:
Could you add some docstring about the logic here? What is `trainer.fit_loop._combined_loader` for those of us who don't know much PL? :p